include/deal.II/base/types.h

```diff
@@ -41,6 +41,12 @@ namespace types
    */
   typedef unsigned int subdomain_id;
 
+  /**
+   * The type used for global indices of vertices.
+   */
+  typedef unsigned long long int global_vertex_index;
+#  define DEAL_II_VERTEX_INDEX_MPI_TYPE MPI_UNSIGNED_LONG_LONG
```

Comment:
What is the reason for introducing a new index type? Intuitively I would think that the vertex index type and the dof index type should match; I imagine quite some confusion in our code if this is not the case.